### DISCUSSION TITLE
QA: Cross OS Build Testing & Documentation

### DISCUSSION THREAD

**Discussion by georgeartem** [dogecoin/dogecoin#3471] [Help needed]
This issue arises from discussion in #3437 to assist with understanding which OS flavors are "in scope" for core developer support and documented as "clean" for new developer onboarding.

Per @patricklodder it is recommended to start with linux distributions that already have documented build guides and work outwards with creating guides and documenting known issues for (hopefully) all builds represented in the Minimum (pre-compiled binary) OS documentation. 

Broadly speaking, for each maintained version, "Core QA" will create a Dockerfile and run a build each time dependencies are updated. 

When something fails "Core QA" will document failures by opening an issue with the Dockerfile for senior members of the core development team to reproduce the errors, warnings etc. and work through a solution.

Alternatively, if "Core QA" is feeling advanced they are free to take on and work through the issue themselves by submitting PRs with support from senior members of the team.

**Comment by patricklodder** [comment]
> which OS flavors are "in scope" for core developer support

"core developer" -> "community", because community > titles

> "Core QA"

Why does this activity need this title/group?

**Comment by georgeartem** [comment]
maybe it does, maybe it doesn't, it's just a label.

**Comment by patricklodder** [comment]
To help **the community** get started: here's a Dockerfile that you can use to test archlinux (rolling distro so it always has rather up-to-date packages):

```Dockerfile
FROM archlinux
WORKDIR /build
RUN pacman -Sy \
 && pacman --noconfirm -S git base-devel boost libevent python db

ARG GITREF=1.15.0-dev
ARG MAKEOPTS=-j8

RUN git clone https://github.com/dogecoin/dogecoin.git \
 && cd dogecoin \
 && git checkout $GITREF \
 && ./autogen.sh \
 && ./configure --without-gui --without-miniupnpc \
 && make $MAKEOPTS \
 && make check
```

running this will attempt to self-compile Dogecoin Core from the `1.15.0-dev` branch (CLI only, so without GUI for this build) with the versions of the packages distributed by Archlinux, which will error out to the bug mentioned in #3454. This is because Archlinux now has boost-1.84, just like the author of the issue, whereas 2 weeks ago when I last tested it (per #3437) it did not. 

This bug is already reported, and a fix is proposed with #3456, which is waiting for review. Once that has been merged, you can run this Dockerfile again, to see if that resolved the issue, or change the cloned url and GITREF variable in the docker file to test the patch while it's still pending, after which you could even leave a comment at the pull request that you tested it and it resolved the issue, which would potentially help getting it through.

By doing this, the community would:

1. Be aware of issues quickly, so that fixes can be proposed.
2. Help other shibes to not run into problems because they are tackled fast
3. Identify any issues with the build guides - in this case the Archlinux build guide.

**Comment by georgeartem** [comment]
I’ve noted somewhere between one and two dozen issues in the last month that appear to be loosely related to this topic.
